js/articles.js

```diff
@@ -0,0 +1,64 @@
+var Article = function(props) {
+  this.title = props.title;
+  this.category = props.category;
+  this.author = props.author;
+  this.authorUrl = props.authorUrl;
+  this.date = props.publishedOn;
+  this.publishedOn = Date.parse(props.publishedOn);
+  this.body = props.body;
+};
+
+blog.sortArticles = function() {
+  blog.articles.sort(function(a, b) {
+    return b.publishedOn-a.publishedOn;
+  });
+};
+
+Article.prototype.timestamp = function( ) {
+  var now = Date.parse(new Date());
+  var msPerMinute = 60 * 1000;
+  var msPerHour = msPerMinute * 60;
+  var msPerDay = msPerHour * 24;
+  var msPerMonth = msPerDay * 30;
+  var msPerYear = msPerDay * 365;
+  var elapsed = now - this.publishedOn;
+  var dayDiff = Math.floor(elapsed / 86400);
+
+  if (elapsed < msPerMinute) {
+    return Math.round(elapsed/1000) + ' seconds ago';
+  }else if (elapsed < msPerHour) {
+    return Math.round(elapsed/msPerMinute) + ' minutes ago';
+  }else if (elapsed < msPerDay ) {
+    return Math.round(elapsed/msPerHour) + ' hours ago';
+  }else if (elapsed < msPerMonth) {
+    return 'approximately ' + Math.round(elapsed/msPerDay) + ' days ago';
+  }else if (elapsed < msPerYear) {
+    return 'approximately ' + Math.round(elapsed/msPerMonth) + ' months ago';
+  }else {
+    return 'approximately ' + Math.round(elapsed/msPerYear) + ' years ago';
```

Comment:
Nice. Consider at some point adding logic to properly address whether a time-period is singular or not e.g. '1 day ago' instead of '1 days ago'.